### ISSUE TITLE
Repeated hover does't work

### ISSUE THREAD

**Issue by lnsight** [teamcapybara/capybara#2156]
## Meta
Capybara Version: 3.13.2

Driver Information (and browser if relevant):
selenium-webdriver 3.141.0
geckodriver-v0.24.0-linux64
firefox-65.0 and later

## Expected Behavior
```ruby
find_link('Label').hover # successful
<going to other page>
<going back>
find_link('Label').hover # successful yet
```
## Actual Behavior

```ruby
find_link('Label').hover # successful
<going to other page>
<going back>
find_link('Label').hover # does nothing (without any errors)
```
## Steps to reproduce

https://gist.github.com/lnsight/1440b63dd3fbbaf4c612a0714342f66b

**Comment by twalpole** [comment]
Do you have an example that reproduces this?  Are you moving to somewhere else before trying to hover again?  This report really doesn't provide me any way to reproduce your issue.  I've created a gist  - https://gist.github.com/twalpole/14041cfa75a3445e6e155a9990c5d19f- that hovers multiple times on 2 elements and works fine.  If you can produce an example that shows your issue then I will take a look at it, otherwise there's not much for me to go on here.

**Comment by lnsight** [comment]
This bug reproduced with firefox 65.0 and later, firefox 64.0.2 works as expected.

**Comment by twalpole** [comment]
The code I linked above works fine for me with FF 65.0.1 - Unless you can provide code that reproduces the issue there is nothing I can do here.

**Comment by lnsight** [comment]
```ruby 
visit article_path
find_link('Settings').hover
click_link 'Edit' # opens edit form
click_link 'Back' # back to show
find_link('Settings').hover
click_link 'Edit' # raises Capybara::ElementNotFound
```

**Comment by twalpole** [comment]
@Insight What HTML am I supposed to run that against? I need to be able to reproduce this in order to do anything. Take the gist I linked to and modify it to show the issue if your app isn’t publicly available.

**Comment by twalpole** [comment]
Closing - will reopen if/when a way to reproduce this is provided

**Comment by lnsight** [comment]
I am reproduced bug in https://gist.github.com/lnsight/1440b63dd3fbbaf4c612a0714342f66b.

**Comment by twalpole** [comment]
@lnsight Thanks - that code does reproduce it for me. As you stated it does appear to be a change in behavior for FF and should probably be reported to the geckodriver project.  Short term It's possible to work around it in your project by forcing the mouse to move somewhere else first (call hover on any other element on the page).  Medium term think Capybara can work around the issue by moving to an offset on the element before moving to its center for the FF driver.  Longer term the best solution would be if geckodriver/FF fixed the behavior

**Comment by twalpole** [comment]
@lnsight Please try the issue_2156 branch to see if it fixes your issue (without breaking anything else)

**Comment by lnsight** [comment]
@twalpole  Thank you for your help. issue_2156 branch fixes the issue. 
But i've found a changing of behavior of #drag_to in FF 65.0.1.
We have ordered list of lines:
```rb
verify_numbers('NUM1', 'NUM2', 'NUM3')
source = find('.number tr', text: 'NUM1').find('.sort-handle')
target = find('.number tr', text: 'NUM3')
source.drag_to(target)
verify_numbers('NUM2', 'NUM3', 'NUM1')
```
Result order is 'NUM2', 'NUM1', 'NUM3'.

**Comment by lnsight** [comment]
I've reported about this issue to geckodriver https://github.com/mozilla/geckodriver/issues/1507.

**Comment by twalpole** [comment]
@Insight If your drag/drop code is using HTML5 drag & drop the change may be because of the change in master Capybara that adds an extra move so JS can determine direction of move. If that's the case then it would be good to know whether or not the current behavior is now correct for the code you're using (and was wrong before), or whether it is now incorrect.  It may also depend on exactly where the drop occurs and sub pixel accuracy determine whether the element should be inserted or appended.  If the change is caused by the Capybara change (check master vs  release) then please create a new issue and provide a way to replicate.  If it's purely a browser version/pixel accuracy issue there may not be anything Capybara can do about it since it may be correct behavior as far as the browser is concerned.

**Comment by twalpole** [comment]
Closed via b7620b9851b5c6bb1830344545f2f1c0455102cb
